framework/PluginsBaseClasses/PostProcessorPluginBase.py

```diff
@@ -0,0 +1,129 @@
+# limitations under the License.
+"""
+Created on March 10, 2021
+
+@author: wangc
+"""
+
+#External Modules---------------------------------------------------------------
+import abc
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+import Files
+from utils import InputData, InputTypes
+from DataObjects import DataObject
+from Databases import Database
+from .PluginBase import PluginBase
+from Models.PostProcessors.PostProcessorInterface import PostProcessorInterface
+#Internal Modules End-----------------------------------------------------------
+
+class PostProcessorPluginBase(PostProcessorInterface, PluginBase):
+  """
+    This class represents a specialized class from which each PostProcessor plugins must inherit from
+  """
+  # List containing the methods that need to be checked in order to assess the
+  # validity of a certain plugin. This list needs to be populated by the derived class
+  _methodsToCheck = ['getInputSpecification', '_handleInput', 'run']
+  entityType = 'PostProcessor'
+
+  ##################################################
+  # Methods for Internal Use
+  ##################################################
+  def createNewInput(self,inputObjs,samplerType,**kwargs):
+    """
+      This function is used to convert internal DataObjects to user-friendly format of data.
+      The output from this function will be directly passed to the "run" method.
+      @ In, inputObjs, list, list of DataObjects
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input.
+          Not used for PostProcessor, and "None" is used during "Step" "PostProcess" handling
+      @ In, **kwargs, dict, is a dictionary that contains the information coming from the sampler,
+          a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}.
+          Not used for PostProcessor, and {'SampledVars':{'prefix':'None'}, 'additionalEdits':{}}
+          is used during "Step" "PostProcess" handling
+      @ Out, inputDs, list, list of data set that will be directly used by the "PostProcessor.run" method.
+    """
+    #### TODO: This method probably need to move to PostProcessor Base Class when we have converted
+    #### all internal PostProcessors to use Dataset
+
+    ## Type 1: DataObjects => Dataset
+    ## Type 2: File => File
+    ## Type 3: HDF5 => ?
+    assert type(inputObjs) == list
+    inputDs = []
+    for inp in inputObjs:
+      if isinstance(inp, Files.File):
+        inputDs.append(inp)
+      elif isinstance(inp, DataObject.DataObject):
+        # Current accept two types: 1) 'dict', 2) 'xrDataset'
+        # Set default to 'dict', this is consistent with current post-processors
+        outType = kwargs.get('outType', 'dict')
+        inputDs.append(inp.asDataset(outType=outType))
+      elif isinstance(inp, Database):
```

Comment:
all these checks should be moved at the validation stage in the Models.py (I thought we already have it. Doesn't it work?). Databases are not possible for PP and they should be checked for all of them at once

Reply:
removed.